plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,477 @@
+version_added: 2.20.0
+description:
+    - "This module allows one to generate or delete GPG private and public keys using GnuPG (gpg)."
+requirements:
+    - gpg >= 2.1
+extends_documentation_fragment:
+    - ansible.builtin.files
+    - community.crypto.attributes
+    - community.crypto.attributes.files
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: none
+options:
+    state:
+        description:
+            - Whether the private and public keys should exist or not, taking action if the state is different from what is stated.
+        type: str
+        default: present
+        choices: [ present, absent ]
+    key_type:
+        description:
+            - Specifies the type of key to create.
+        type: str
+        choices: ['RSA', 'DSA', 'ECDSA', 'EDDSA']
+    key_length:
+        description:
+            - For non-ECC keys, this specifies the number of bits in the key to create.
+            - For RSA keys, the minimum is V(1024), the maximum is V(4096), and the default is V(3072).
+            - For DSA keys, the minimum is V(768), the maximum is V(3072), and the default is V(2048).
+            - As per gpg's behavior, values below the allowed ranges will be set to the respective defaults, and values above will saturate at the maximum.
+            - For ECC keys, this parameter will be ignored.
+        type: int
+    key_curve:
+        description:
+            - For ECC keys, this specifies the curve used to generate the keys.
+            - EDDSA keys only support the V(ed25519) curve and they can only be generate using said curve.
+            - For ECDSA keys, the default is V(brainpoolP512r1).
+            - For non-ECC keys, this parameter with be ignored.
+        type: str
+        choices: ['nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1', 'ed25519']
+    key_usage:
+        description:
+            - Specifies usage(s) for key.
+            - V(cert) is given to all primary keys regardess, however can be used to only give V(vert) usage to a key.
+            - If not usage is specified, the valid usages for the given key type with be assigned.
+        type: list
+        elements: str
+        choices: ['encrypt', 'sign', 'auth', 'cert']
+    subkeys:
+        description:
+            - List of subkeys with their own respective key types, lengths, curves, and usages.
+            - Similar to O(key_type), O(key_length), O(key_curve), and (key_usage).
+            - Supports ECDH and ELG keys.
+            - For both ECDH and ELG keys, the only supported usage is V(encrypt).
+            - For ECDH keys, the default curve is V(brainpoolP512r1).
+            - ECDH keys also support the V(cv25519) curve.
+            - For ELG keys, the minimum length is V(1024) bits, the maximum length is V(4096) bits, and the default length is V(3072) bits.
+        type: list
+        elements: dict
+        options:
```

Comment:
In `DOCUMENTATION` it's `suboptions`, not `options`:
```suggestion
        suboptions:
```